frontend/packages/pipelines-plugin/src/components/pipelines/pipeline-builder/update-utils.ts

```diff
@@ -439,21 +539,74 @@ export const applyChange = (
   op: UpdateOperation,
 ): CleanupResults => {
   const { type, data } = op;
-  const { tasks, listTasks } = taskGroup;
+  const { tasks, listTasks, finallyTasks, finallyListTasks } = taskGroup;
 
   switch (type) {
     case UpdateOperationType.ADD_LIST_TASK:
-      return addListNode(tasks, listTasks, data as UpdateOperationAddData);
+      return addListNode(
+        tasks,
+        listTasks,
+        data as UpdateOperationAddData,
+        finallyTasks,
+        finallyListTasks,
```

Comment:
Hmmm... concerned that we ended up adding a bunch of properties that are just pass-thru... 

Maybe wrap the return statements that don't use a specific set of data?

Something like:
```
return withUnusedData(addListNode(tasks, listTasks, data as UpdateOperationAddData))
```
Where `withUnusedData` simply just adds it onto the end of the data:
```
const withUnusedData = (updatedValues) => ({ tasks, listTasks, finallyTasks, finallyListTasks, ...updatedValues })
// switch statement
```

Ideally this would allow us to only modify what we need (and the following comment will allow all the data to still be available)... power of spread will give us a nice clean fallback without stepping on anything / being redundant.